org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/bundle/BundleSource.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.osgi.test.junit5.bundle;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.osgi.test.common.annotation.Property;
+
+@Target({
+	ElementType.ANNOTATION_TYPE, ElementType.METHOD
+})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@ArgumentsSource(BundleArgumentsProvider.class)
+public @interface BundleSource {
+
+	/**
+	 * Optional SymbolicNameFilter used to filter Bundles by SymbolicName using
+	 * the an regex patterns.
+	 *
+	 * @return The symbolicNameFilter Strings.
+	 */
+	String[] symbolicNameFilter() default {};
+
+	/**
+	 * Optional stateFilter used to filter Bundles by state.
+	 *
+	 * @return The states.
+	 */
+	int[] stateFilter() default {};
+
+	/**
+	 * Optional headerFilter used to filter Bundles by state.
+	 *
+	 * @return The states.
+	 */
+	Property[] headerFilter() default {};
```

Comment:
`String[][]` is not valid. Only one dimensional arrays can be used.  

Why is this not a filter string like in ServiceSource? Then you can filter over the bundle headers.

`Property` is not correct since you are really filtering and the type information is `Property` does not make sense here.